dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java

```diff
@@ -589,4 +589,34 @@ public void Test_Render_Menu_Items_On_Edit_Mode_Page_Mode(final PageMode pageMod
     public static Object[] editAndPreviewModes() {
         return new Object[]{PageMode.EDIT_MODE,PageMode.PREVIEW_MODE};
     }
+
+    /**
+     * Method to test: NavTool.getNav
+     * Given scenario: get navigation for a folder where it contains a multilingual page, only request the nav for the
+     * non-default language
+     * Expected result: getting the navigation should only return the page of the non-default language
+     * @throws Exception exception
+     */
+    @Test
+    public void test_getNav_multilingualPage_shouldOnlyReturnOne() throws Exception {
+        //Create Folder
+        folder = new FolderDataGen().site(site).showOnMenu(true).nextPersisted();
+
+        //New template
+        final Template template = new TemplateDataGen().nextPersisted();
+
+        //Create a multilingual page with show on menu true
+        final HTMLPageAsset pageAssetEng = new HTMLPageDataGen(folder, template).showOnMenu(true).languageId(1).nextPersisted();
+        pageAssetEng.setIndexPolicy(IndexPolicy.FORCE);
+        APILocator.getContentletAPI().publish(pageAssetEng, user, false);
+        final Contentlet pageAssetSpa = APILocator.getContentletAPI().checkout(pageAssetEng.getInode(),user,false);
+        pageAssetSpa.setLanguageId(spanishLanguage.getId());
+        APILocator.getContentletAPI().checkin(pageAssetSpa,user,false);
+        APILocator.getContentletAPI().publish(pageAssetSpa,user,false);
+
+        //get the nav
+        final NavResult navResult = new NavTool().getNav(site,folder.getPath(),spanishLanguage.getId(),user);
+        assertNotNull(navResult);
+        assertEquals(1,navResult.getChildren().size());
```

Comment:
Shouldn't we assert that the element is the one we are expected? not only the size of the list

Reply:
done